corehq/apps/enterprise/interface.py

```diff
@@ -0,0 +1,206 @@
+from django.db.models.aggregates import Count
+from corehq.apps.reports.generic import GenericTabularReport
+from corehq.apps.enterprise.dispatcher import EnterpriseInterfaceDispatcher
+from corehq.apps.reports.datatables import DataTablesColumn, DataTablesHeader
+from corehq.apps.accounting.filters import DateCreatedFilter, NameFilter
+from corehq.apps.sms.models import INCOMING, OUTGOING
+
+from corehq.apps.smsbillables.filters import (
+    DateSentFilter,
+    GatewayTypeFilter,
+    HasGatewayFeeFilter,
+    ShowBillablesFilter,
+)
+from corehq.apps.enterprise.filters import EnterpriseDomainFilter
+from corehq.apps.accounting.models import (
+    BillingAccount,
+    Subscription
+)
+from corehq.apps.smsbillables.models import (
+    SmsBillable,
+)
+
+from couchexport.models import Format
+from dimagi.utils.dates import DateSpan
+
+
+class EnterpriseSMSBillablesInterface(GenericTabularReport):
```

Comment:
I don't have enough knowledge to provide other options, but I'm assuming we've exhausted all other options that would avoid duplicating a report?

Reply:
The duplication is, unfortunately, a result of the current reporting framework. avoiding it would require some major refactoring I fear the inheritance patterns you are forced into here might obscure potential security risks

Reply:
It's a trade-off either way. With this much duplicated code, if a bug was discovered in `SMSBillablesInterface`, we would have to hope that a developer knew to also fix that bug in the enterprise class. Yes, identifying the security issues takes more time now, but the advantage of doing this is that we avoid more costly bugs in the future.

I don't know how much enterprise reports are under the umbrella of reports in general, but as reports are my vertical at the moment, and I'm working through some security issues with reports anyway, I'd be happy to run through this to see what changes would be needed to prevent duplication.